tools/nni_gpu_tool/gpu_metrics_collector.py

```diff
@@ -48,11 +49,13 @@ def main(argv):
     with open(os.path.join(metrics_output_dir, "gpu_metrics"), "w") as outputFile:
         pass
     os.chmod(os.path.join(metrics_output_dir, "gpu_metrics"), 0o777)
-    cmd = 'nvidia-smi -q -x'
+    cmd = 'nvidia-smi -q -x'.split()
     while(True):
         try:
-            smi_output = subprocess.check_output(cmd, shell=True)
+            smi_output = subprocess.check_output(cmd)
             parse_nvidia_smi_result(smi_output, metrics_output_dir)
```

Comment:
there is try catch within `parse_nvidia_smi_result`

Reply:
and it uses 'a' in `open`

Reply:
This is to catch `subprocess.check_output` when `nvidia-smi` not exist